letsApply/Controllers/TabBarViewController/HomeScreenViewController/HomeViewModel/HomeViewModel.swift

```diff
@@ -4,51 +4,74 @@
 //
 //  Created by Reuben Simphiwe Kuse on 2024/12/16.
 //
+
 import Foundation
 import FirebaseFirestore
+import FirebaseStorage
+import UIKit
+import FirebaseAuth
 
 class HomeViewModel {
+    
+    // MARK: - Properties
+    
     private let firestoreService = FirestoreService()
-    var userProfile: UserProfile?
-    var allJobs: [Job] = []           // All fetched jobs
-    var tailoredJobs: [Job] = []      // Filtered or tailored jobs for UI
+    private let storage = Storage.storage()
+    
+    // Categories to display on Home Screen
+    let categories: [JobCategory] = [
+        JobCategory(title: "Applied Jobs", icon: "doc.text.fill"),
+        JobCategory(title: "Saved Jobs", icon: "bookmark.fill"),
+        JobCategory(title: "Recommended", icon: "star.fill"),
+        JobCategory(title: "Nearby Jobs", icon: "location.fill")
+    ]
     
-    var onDataUpdated: (() -> Void)?
+    // MARK: - Fetch User Profile
     
-    func fetchUserProfile(uid: String) {
-        firestoreService.fetchUserProfile(uid: uid) { [weak self] result in
+    func fetchUserProfile(completion: @escaping (UserProfile) -> Void) {
+        guard let uid = Auth.auth().currentUser?.uid else { return }
```

Comment:
How would you want this to complete when the guard statement fails. For now it's just returning but it's not calling the completion block